### DISCUSSION TITLE
Handling cookie string values might be slightly off in CookiesJar.update_cookies

### DISCUSSION THREAD

**Discussion by lucendio** [aio-libs/aiohttp#8296] [General]
I kindly ask for some feedback on the following matter. It looks like [LOC 184](https://github.com/aio-libs/aiohttp/blob/54e13b0a2a2ac52cedaaa3837f9e93de220a423f/aiohttp/cookiejar.py#L184) is not correct:

```diff
if not isinstance(cookie, Morsel):
    tmp = SimpleCookie()
    tmp[name] = cookie  # type: ignore[assignment]
-   cookie = tmp[name]
+   cookie = tmp
```
I presume this condition is supposed to handle the case of `cookie` being a `str`. What do you guys think?

**Comment by Dreamsorcerer** [comment]
I'm not sure what you think is wrong with that, but how about trying to write a test to demonstrate the issue?

**Reply by lucendio** [reply]
before putting some effort into code writing I was looking for another pair of eyes to look at this thing, which is why I chose the discussions section.

It looks like the downstream code (below the condition block) assumes `cookie` to be of type `Morsel` (or some dict-like value for that matter), but with the line `cookie = tmp[name]` the value of `cookie` does not change due to the previous line `tmp[name] = cookie`. So, instead of `cookie` becoming an instance of `SimpleCookie` it stays the same. which in this case (not of type `Morsel`) is probably the actual string value of that cookie.

Does this help to better understand what I'm trying to convey?

**Reply by Dreamsorcerer** [reply]
> but with the line cookie = tmp[name] the value of cookie does not change due to the previous line tmp[name] = cookie.

getters and setters don't always return/receive the same type. ;)

Try actually running the code, even just writing that code into an interpreter. I'm pretty sure you'll get a Morsel.

**Reply by lucendio** [reply]
Thank you! That was very helpful! I'll go ahead an close this thread.
